src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -143,25 +110,31 @@ private void generateXmlTest(ISuite suite,
       // sorted, we don't need to sort them again.
       //
       List<ITestNGMethod> result = Lists.newArrayList();
+      Set<ITestNGMethod> relevantConfigs = Sets.newHashSet();
```

Comment:
Minimal change is a good habit IMO but improving current code too.
A good compromise is doing it in 2 steps: 1 commit for fixing the issue with minimal changes + 1 commit to improve/refactor